subsys/bluetooth/services/ots/ots_dir_list.c

```diff
@@ -0,0 +1,226 @@
+	sys_put_le16(ots->dir_list->net_buf.len, ots->dir_list->net_buf.data);
+
+	/* Update Directory Listing Object metadata size */
+	ots->dir_list->dir_list_obj->metadata.size.cur = ots->dir_list->net_buf.len;
+}
+
+static void dir_list_encode(struct bt_ots *ots)
+{
+	struct bt_gatt_ots_object *obj;
+	int err;
+
+	err = bt_gatt_ots_obj_manager_first_obj_get(ots->obj_manager, &obj);
+
+	__ASSERT(first_obj == ots->dir_list->dir_list_obj,
+		 "Expecting first object to be the Directory Listing Object");
+
+	/* Init with len = 0 to reset data */
+	net_buf_simple_init_with_data(&ots->dir_list->net_buf,
+				      ots->dir_list->_content,
+				      sizeof(ots->dir_list->_content));
+	ots->dir_list->net_buf.len = 0;
+
+	do {
+		dir_list_object_encode(obj, &ots->dir_list->net_buf);
+
+		err = bt_gatt_ots_obj_manager_next_obj_get(ots->obj_manager, obj, &obj);
+	} while (!err);
+
+	/*re-encode the Directory Listing Object size with the is new size*/
+	sys_put_le16(ots->dir_list->net_buf.len, ots->dir_list->net_buf.data);
+
+	/* Update Directory Listing Object metadata size */
+	ots->dir_list->dir_list_obj->metadata.size.cur = ots->dir_list->net_buf.len;
+}
+
+void bt_ots_dir_list_obj_selected(struct bt_ots *ots)
+{
+	if (ots->dir_list->dir_list_obj != ots->cur_obj) {
+		/* We only need to update the object directory listing if it is currently selected,
+		 * as we otherwise only create it when it is selected.
+		 */
+		return;
+	}
+
+	dir_list_encode(ots);
+}
+
+void bt_ots_dir_list_obj_init(struct bt_ots *ots)
+{
+	struct bt_gatt_ots_object *dir_list_obj;
+	int err;
+	static char *dir_list_obj_name = CONFIG_BT_OTS_DIR_LIST_OBJ_NAME;
+
+	if (!ots->dir_list) {
+		for (int i = 0; i < ARRAY_SIZE(dir_lists); i++) {
+			if (!dir_lists[i].dir_list_obj) {
+				ots->dir_list = &dir_lists[i];
+			}
+		}
+	}
+
+	__ASSERT(ots->dir_list,
+		 "Could not assign Directory Listing Object to OTS instance %p", ots);
+
+	__ASSERT(strlen(dir_list_obj_name) <= UINT8_MAX,
+		 "BT_OTS_DIR_LIST_OBJ_NAME shall be less than 255 octets");
+
+	err = bt_gatt_ots_obj_manager_obj_add(ots->obj_manager, &dir_list_obj);
+
+	__ASSERT(!err, "Could not add Directory Listing Object for object manager %p", obj_man);
+
+	memset(&dir_list_obj->metadata, 0, sizeof(dir_list_obj->metadata));
+	dir_list_obj->metadata.name = dir_list_obj_name;
+	dir_list_obj->metadata.size.alloc = sizeof(ots->dir_list->_content);
+	dir_list_obj->metadata.type.uuid.type = BT_UUID_TYPE_16;
+	dir_list_obj->metadata.type.uuid_16.val = BT_UUID_OTS_DIRECTORY_LISTING_VAL;
+	BT_OTS_OBJ_SET_PROP_READ(dir_list_obj->metadata.props);
+
+	ots->dir_list->dir_list_obj = dir_list_obj;
+
+	/* Set size in dir_list_encode */
+	dir_list_encode(ots);
+}
+
+int bt_ots_dir_list_content_get(struct bt_ots *ots, uint8_t **data,
+				uint32_t len, uint32_t offset)
+{
+	if (offset >= ots->dir_list->net_buf.len) {
+		*data = NULL;
+		return 0;
+	}
+
+	*data = ots->dir_list->net_buf.data;
```

Comment:
What about the offset?
This returns the same pointer, no matter what offset parameter is given in.

Reply:
Hah, yeah that's something important missing :D